### ISSUE TITLE
Clicking on links in html emails

### ISSUE THREAD

**Issue by hugoroy** [astroidmail/astroid#528]
Does it work for you when reading html emails? I can't seem to be able to click on links (also the pointer does detect a link).

**Comment by gauteh** [comment]
It should work. Is the click registered in the logs?

tir. 24. jul. 2018 kl. 15:29 skrev Hugo Roy <notifications@github.com>:

**Comment by gauteh** [comment]
Still an issue? Note that anything using Javascript will not work. The GitHub links in the HTML parts work for me.

**Comment by hugoroy** [comment]
Did not have time to investigate. Will do this weekend!

Le 27 juillet 2018 09:17:26 GMT+02:00, Gaute Hope <notifications@github.com> a écrit :
>Still an issue? Note that anything using Javascript will not work. The
>GitHub links in the HTML parts work for me.

-- 
Envoyé de mon appareil mobile
Sent from my mobile device

**Comment by hugoroy** [comment]
To be more precise,

 - I can hover a link and it gets detected as such (pointer changes to a hand)
 - I can right click on a link and have a link-specific menu
 - I can "copy" the target link from this menu
 - If I click to open the link or open in a new window from this menu, nothing happens

I don't  see anything in the logs for any of the above. I have tried `coredumpctrl astroid gdb`  but I get "Unknown operation astroid."

**Comment by hugoroy** [comment]
↪ Gaute Hope / 2018-07-29, 18:28 (CEST):
> Also make sure to turn on debug level log messages

Interesting, I wanted to look into how to do that, and here's what happened:

```

hugo@xps ~ % astroid --help
  -h [ --help ]         print this help message
  -c [ --config ] arg   config file, default: $XDG_CONFIG_HOME/astroid/config
  -n [ --new-config ]   make new default config, then exit
  -m [ --mailto ] arg   compose mail with mailto url or address
  --no-auto-poll        do not poll automatically
  --disable-log         disable logging
  --log-stdout          log to stdout regardless of configuration
  --start-polling       indicate that external polling (external notmuch db R/W
                        operations) starts
  --stop-polling        indicate that external polling stops
  --refresh arg         refresh messages changed since lastmod
  --disable-plugins     disable plugins

[1]    1188 segmentation fault (core dumped)  astroid --help

```


> 
> søn. 29. jul. 2018 kl. 18:28 skrev Gaute Hope <eg@gaute.vetsj.com>:
> 
>> Does left click without using menu work?


No, nothing happens when I do that, and nothing appears in the terminal. (But maybe I didn't specify the right debug log level).

**Comment by gauteh** [comment]
Hugo Roy writes on July 29, 2018 18:40:
> ↪ Gaute Hope / 2018-07-29, 18:28 (CEST):
>> Also make sure to turn on debug level log messages
> 
> Interesting, I wanted to look into how to do that, and here's what happened:
> 
> ```
> [1]    1188 segmentation fault (core dumped)  astroid --help
> 
> ```

Yeah -- that happens if you already have an instance of astroid running. 
Should fix that...


>> 
>> søn. 29. jul. 2018 kl. 18:28 skrev Gaute Hope <eg@gaute.vetsj.com>:
>> 
>>> Does left click without using menu work?
> 
> 
> No, nothing happens when I do that, and nothing appears in the terminal. (But maybe I didn't specify the right debug log level).
> 

You either need to get latest master for a command line arg, or use the 
config to set it: https://github.com/astroidmail/astroid/wiki/Configuration-Reference#undocumented-options

**Comment by hugoroy** [comment]
↪ Gaute Hope / 2018-07-29, 18:44 (CEST):
> Yeah -- that happens if you already have an instance of astroid running. 
> Should fix that...

Indeed, I don't get that when astroid does not already run! :)


> You either need to get latest master for a command line arg, or use the 
> config to set it: https://github.com/astroidmail/astroid/wiki/Configuration-Reference#undocumented-options

Which number should I put?

**Comment by gauteh** [comment]
Hugo Roy writes on July 29, 2018 18:48:
> ↪ Gaute Hope / 2018-07-29, 18:44 (CEST):
>> You either need to get latest master for a command line arg, or use the 
>> config to set it: https://github.com/astroidmail/astroid/wiki/Configuration-Reference#undocumented-options
> 
> Which number should I put?

The string "debug".

**Comment by hugoroy** [comment]
↪ Gaute Hope / 2018-07-29, 18:49 (CEST):
> Hugo Roy writes on July 29, 2018 18:48:
>> ↪ Gaute Hope / 2018-07-29, 18:44 (CEST):
>>> You either need to get latest master for a command line arg, or use the 
>>> config to set it: https://github.com/astroidmail/astroid/wiki/Configuration-Reference#undocumented-options
>> 
>> Which number should I put?
> 
> The string "debug".
> 

Like this?

```
{
    "astroid": {
        "log": {
            "level": "debug",
            "syslog": "debug",
            "stdout": "debug"
        },
```

**Comment by gauteh** [comment]
Hugo Roy writes on July 29, 2018 18:53:
> ↪ Gaute Hope / 2018-07-29, 18:49 (CEST):
>> Hugo Roy writes on July 29, 2018 18:48:
>>> ↪ Gaute Hope / 2018-07-29, 18:44 (CEST):
>>>> You either need to get latest master for a command line arg, or use the 
>>>> config to set it: https://github.com/astroidmail/astroid/wiki/Configuration-Reference#undocumented-options
>>> 
>>> Which number should I put?
>> 
>> The string "debug".
>> 
> 
> Like this?
> 

Like this:

```
{
    "astroid": {
        "log": {
            "level": "debug",
            "syslog": "true",
            "stdout": "true"
        },
```

**Comment by hugoroy** [comment]
OK. So: I didn't see anything in the logs.

However, I realise that it works on some links, but not others... A bit hard to be more precise at this point. My impression is that it used to work better before in this regard. But this is just an impression...

Links in Github html footers seem to work.

**Comment by gauteh** [comment]
Do these links use Javascript? I get the same issues for some links.

**Comment by hugoroy** [comment]
Hi, finally - here's an example email where I can't click on the links.

[Les nouveaut_s SNCF et OUIGO_ et de beaux voyages _ venir.zip](https://github.com/astroidmail/astroid/files/3113327/Les.nouveaut_s.SNCF.et.OUIGO_.et.de.beaux.voyages._.venir.zip)

**Comment by serussell** [comment]
Same behavior here; absolutely nothing in printed in the logs (either console or 'Z') when the link is clicked.  Here's a truncated link in the source of an email that isn't working.  It's weird HTML, but it isn't javascript.  The link is truncated for privacy reasons; it's just a very long base64 encoded packet, no other parameters. When I copy the link with right-click menu and paste into a browser, it works:

```
<span id=3D"unsubscribe-link">|<a href=3D"http://sg.theladders.com/mpss/c/I=
wE/X1sVAA/t.2r2/BH0NjXtaRb6e6swM1wVATA/h8/Grv45......." style=3D"color: #00c0d1; text-decoratio=
n: none;" target=3D"_blank">=C2=A0Unsubscribe</a></span>
```

**Comment by JayVii** [comment]
Is there any progress for this issue? This issue usually happens for mail-digest from my university. The URLs have an odd form and the HTML in the mail is rather "unclean":

```
                        <tr style=3D"background-color: #cdcdcd">
                            <td style=3D"padding: 5px; font-family: Helveti=
ca, Arial, sans-serif;">
                                <a href=3D"http://www.uni-trier.de/index.ph=
p?id=3D69768&amp;tx_zimkannouncements_zimkannouncementsdisplay[controller]=
=3DMessage&amp;tx_zimkannouncements_zimkannouncementsdisplay[action]=3Dshow=
Public&amp;tx_zimkannouncements_zimkannouncementsdisplay[message]=3D2517" t=
arget=3D"_blank" style=3D"text-decoration: none; color: #134C75;"><b>0,5 Cr=
edits f=C3=BCr Online-Studie MEN20D</b></a><br>
                                Kurze Online-Studie zur Wahrnehmung von M=
=C3=A4nnern (MEN20D)!<br>
                            </td>
                        </tr>
```

I can see, that the user above me has a `target=3D"_blank"` inside the `<a></a>` tag as well. Could it either be this specific parameter or `target="_blank"` in general?

**Comment by hugoroy** [comment]
This could be the root cause of the issue indeed!

**Comment by Lattitude75** [comment]
@hugoroy Has anything happened with this issue? It still affects me and is very troubling as I am not able to open links in the Html mode and the plain mode is not exactly readable when the email has many links.

**Comment by mreppen** [comment]
I can confirm that I only experience this with `target="_blank"` links.

@Lattitude75 In case you hadn't thought of it: you can right click and copy the link, then paste it in the browser. Comes with the bonus of seeing the full link as well.

**Comment by Lattitude75** [comment]
@mreppen Yes, I have realised that. But, that is too lengthy each time I want to open a link. I copy it, then go to a browser open a new tab and then paste it. This is too lengthy a process and is very time consuming when you want to open multiple links.

**Comment by mreppen** [comment]
I agree. I did not mean it as a solution, but as a workaround. I wish I had more time on my hands to take a look myself, but that will not happen in months.
